### PR TITLE
Fix for difficulty filtering double-negative, min + max each optional

### PR DESCRIPTION
**Pull request by willccbb**
also removing unused ThinkParser

### CODE REVIEW

File: src/prime_rl/environments/registry.py [hunk 1]
```diff
@@ -44,24 +44,21 @@ def correct_answer_reward_func(completion, answer, **kwargs) -> float:
...
         lambda x: {"question": x["prompt"], "info": json.loads(x["verification_info"]), "task": "simple-math"}
     )
     solve_rate_field = env_args.get("solve_rate_field", None)
-    if solve_rate_field is not None:
-        min_solve_rate = env_args["min_solve_rate"]
-        max_solve_rate = env_args["max_solve_rate"]
-        train_dataset = train_dataset.filter(
-            lambda x: x[solve_rate_field] >= min_solve_rate and x[solve_rate_field] <= max_solve_rate
-        )
+    if solve_rate_field:
```
Comment (samsja) [inline src/prime_rl/environments/registry.py]:
```suggestion
    if solve_rate_field is not None:
```

File: src/prime_rl/environments/registry.py [hunk 2]
```diff
@@ -44,24 +44,21 @@ def correct_answer_reward_func(completion, answer, **kwargs) -> float:
...
+        min_solve_rate = env_args.get("min_solve_rate", None)
+        max_solve_rate = env_args.get("max_solve_rate", None)
+        if min_solve_rate is not None:
+            train_dataset = train_dataset.filter(lambda x: x[solve_rate_field] >= min_solve_rate)
+        if max_solve_rate is not None:
+            train_dataset = train_dataset.filter(lambda x: x[solve_rate_field] <= max_solve_rate)
     train_dataset = train_dataset.remove_columns(["prompt", "verification_info"])
 
-    parser = vf.ThinkParser(extract_fn=extract_boxed_answer)
```
Comment (samsja) [inline src/prime_rl/environments/registry.py:63]: so we don't have think parser anymore ? How is it working rn for the reward and thinking ?

Comment (willccbb) [reply src/prime_rl/environments/registry.py:63]: ThinkParser wasn't actually being used before haha, Parser is only used in multi-turn or if you explicitly reference it in reward functions

**Review by mikasenghaas — APPROVED** [review]
looks good
